Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -162,15 +162,13 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
         // changed, PackagePlugin will need to change as well (but no plugins need to change).
         var command = [compiledExec.pathString]
         command += [String(decoding: input, as: UTF8.self)]
-        
-        #if os(macOS)
-        // If enabled, use sandbox-exec on macOS.  This provides some safety against arbitrary code execution when invoking plugin scripts.
-        // <rdar://40235432> tracks implementing sandboxes for other platforms.
+
+        // If enabled, run command in a sandbox.
+        // This provides some safety against arbitrary code execution when invoking the plugin.
+        // We only allow the permissions which are absolutely necessary.
         if self.enableSandbox {
-            let profile = macOSSandboxProfile(toolsVersion: toolsVersion, writableDirectories: writableDirectories)
-            command = ["/usr/bin/sandbox-exec", "-p", profile] + command
+            command = Sandbox.apply(command: command, writableDirectories: writableDirectories)
```

Comment:
This does assume that the way a sandbox is applied is by modifying the command line, which happens to be true for `sandbox-exec`.  This is fine for now but a more general way to do this would be to have this functionality on the launcher, and to have a sandbox config struct with the writable directories and strictness, etc, and then pass that plus the command to whatever is launching the command.  Again, fine for now, just future thoughts.